### DISCUSSION TITLE
Problem with git push

### DISCUSSION THREAD

**Discussion by vincent-mailhot** [community/community#51177] [Repositories] [answered]
### Select Topic Area

Question

### Body

Hi everyone, I'm trying to push a commit with "git push" and it's not working. The error message is "remote: permission ..myurl... denied to vmailhot" and "fatal: unable to acess ...myurl... : the requested URL returned error: 403.
I hope you can help me, thank you

**Comment by bbkx226** [comment]
Make sure that you are using the correct username and email to authenticate with the Git repository. You can try re-entering your credentials or resetting your password if necessary.

To do that, you can type
```
$ git config --global user.name <Your GitHub name>
```
Then,
```
$ git config --global user.email <Your GitHub Email>
```

2. Verify that you have the necessary permissions to push to the repository, or you have spelled the Git URL wrong

3. Try pushing again using the command "git push -u origin <branch-name>", let's say
```
git push -u origin main
```

**Reply by airtower-luna** [reply]
The `user.name` and `user.email` setting have absolutely nothing to do with authentication, this won't help. They affect *only* what's recorded in commits you make as author name and email.

**Reply by bbkx226** [reply]
> The `user.name` and `user.email` setting have absolutely nothing to do with authentication, this won't help. They affect _only_ what's recorded in commits you make as author name and email.

I see. I felt like I met the problem before and solved it using the command haha might be coincidence

**Comment by vincent-mailhot** [comment]
Tank you for your help. When I'm trying to push the commit to the repository, I'm not prompt for my username and password. I think it's because I'm using a github codespace. I'm aleeady loged in with my github account.

**Reply by bbkx226** [reply]
No problem, friend. If my response is helpful, feel free to mark it as the answer~

**Comment by airtower-luna** [comment]
If you're not asked for authentication that means it's cached *somewhere*. Unfortunately I can't say where that is in a codespace, but that's what you'll need to look for.

**Reply by vincent-mailhot** [reply]
I tried to look on google what can be the cause. The only solutions that I saw were about SSH keys. Is SSH key can help me about git push?

**Reply by airtower-luna** [reply]
You can push using Git over SSH using an SSH key for authentication, or over HTTPS with a token. Either should work if configured correctly. I *guess* that Codespaces would use HTTPS, but I don't actually know.

**Comment by vincent-mailhot** [comment, marked as answer]
Thank you all for your help. I finnaly found a solution, I used instead git desktops and I succeed to push my work properly.

**Comment by ARolfeUAT** [comment]
This is a lame solution and shame on GitHub for not providing an actual fix. I went round and round with this same problem, creating tokens with all the permissions, updating PyCharm, etc. etc. While this solution works, it's almost as if they're trying to force developers to use it. Shame!
